File: web/src/app/shared/Models/forum.ts
export class Forum {
    Id: number = 0;
    SubjectId: number = 0;
    ForumName: string = '';
    Details: string = '';
    CreateDate: Date = new Date();
}
export class ForumModel {
    id: number = 0;
    subjectId: number = 0;
    forumName: string = '';
    details: string = '';
    createDate: Date = new Date();
}
export class Discussion {
    id: number = 0;
    forumId: number = 0;
    discussName: string = '';
    details: string = '';
    createDate: Date = new Date();
}

export class DiscussionModel {
    Id: number = 0;
    ForumId: number = 0;
    DiscussName: string = '';
    Details: string = '';
    CreateDate: Date = new Date();
    UserId: string = '';
    FullName: string = '';
    Email: string = '';
}

File: E-LearnignWebAPI/ElearningBO/E-Learning/Forum.cs
using ElearningBO.UserAuthentication;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ElearningBO.E_Learning
{
    public class Forum
    {
        [Key]
        public int Id { get; set; }

        [ForeignKey("Subject")]
        public int SubjectId { get; set; }
        public string ForumName { get; set; }
        public string Details { get; set; }
        public DateTime CreateDate  {get; set; }
        public bool isDelete { get; set; }
        public virtual Subject Subject { get; set; }

    }
    public class Discussion
    {
        [Key]
        public int Id { get; set; }

        [ForeignKey("Forum")]
        public int ForumId { get; set; }
        public ApplicationUser User { get; set; }
        public string DiscussName { get; set; }
        public string Details { get; set; }
        public DateTime CreateDate { get; set; }
        public bool isDelete { get; set; }
        public virtual Forum Forum { get; set; }
    }
    public class Answer
    {
        [Key]
        public int Id { get; set; }

        [ForeignKey("Discussion")]
        public int DiscussId { get; set; }
        public ApplicationUser User { get; set; }
        public string Details { get; set; }
        public DateTime CreateDate { get; set; }
        public bool isDelete { get; set; }
        public int Reply { get; set; }
        public virtual Discussion Discussion { get; set; }
    }
}
